fork_choice/confirmation-rule.md

```diff
@@ -0,0 +1,647 @@
+        if 2 * total_active_balance <= (ffg_support_for_checkpoint - min_ffg_support_slash_th) * 3:
+            return 100 // 3
+
+    """
+    Case 2: min_ffg_support_slash_th > total_active_balance * ffg_confirmation_score / 100
+
+    2 / 3 * total_active_balance <= \
+        ffg_support_for_checkpoint - \
+        total_active_balance * ffg_confirmation_score / 100 + \
+        (1 - ffg_confirmation_score / 100) * remaining_ffg_weight
+    """
+    ffg_confirmation_score = (
+        100 * (3 * (ffg_support_for_checkpoint + remaining_ffg_weight) - 2 * total_active_balance) //
+        (3 * (remaining_ffg_weight + total_active_balance))
+    )
+    assert ffg_support_for_checkpoint * 100 > total_active_balance * ffg_confirmation_score
+    return max(ffg_confirmation_score, -1)
+```
+
+#### `get_ffg_confirmation_score_previous_epoch`
+
+```python
+def get_ffg_confirmation_score_previous_epoch(total_active_balance: int,
+                                              ffg_support_for_checkpoint: int,
+                                              min_ffg_support_slash_th: int) -> int:
+    assert min_ffg_support_slash_th <= ffg_support_for_checkpoint
+    """
+    Return the max possible ffg_confirmation_score such that:
+    2 / 3 * total_active_balance <= \
+        ffg_support_for_checkpoint - \
+        min(min_ffg_support_slash_th, total_active_balance * ffg_confirmation_score / 100)
+    """
+
+    """
+    Case 1: min_ffg_support_slash_th <= total_active_balance * ffg_confirmation_score / 100
+
+    2 / 3 * total_active_balance <= \
+        ffg_support_for_checkpoint - min_ffg_support_slash_th
+    """
+    if 2 * int(total_active_balance) <= (ffg_support_for_checkpoint - min_ffg_support_slash_th) * 3:
+        return 100 // 3
+
+    """
+    Case 2: min_ffg_support_slash_th > total_active_balance * ffg_confirmation_score / 100
+
+    2 / 3 * total_active_balance <= \
+        ffg_support_for_checkpoint - total_active_balance * ffg_confirmation_score / 100
+    """
+    ffg_confirmation_score = (
+        (300 * (ffg_support_for_checkpoint) - 200 * int(total_active_balance)) //
+        (3 * (total_active_balance))
+    )
+    return max(ffg_confirmation_score, -1)
+```
+
+#### `get_ffg_confirmation_score`
+
+```python
+def get_ffg_confirmation_score(store: Store, 
+                               confirmation_slashing_threshold: int,
+                               block_root: Root) -> int:
+
+    current_epoch = get_current_store_epoch(store)
+
+    block = store.blocks[block_root]
+    block_epoch = compute_epoch_at_slot(block.slot)
+
+    # This function is only applicable to current and previous epoch blocks
+    assert current_epoch in [block_epoch, block_epoch + 1]
+
+    total_active_balance = int(get_total_active_balance_for_block_root(store, block_root))
+
+    ffg_support_for_checkpoint = int(get_ffg_support(store, block_root))
+
+    min_ffg_support_slash_th = min(ffg_support_for_checkpoint, confirmation_slashing_threshold)
```

Comment:
IIUC, ffg_support_for_checkpoint is a total balance voted in support of a checkpoint, while confirmation_slashing_threshold is a portion of stake that adversary is willing to sacrifice.

Shouldn't it be?
```suggestion
    min_ffg_support_slash_th = min(ffg_support_for_checkpoint, confirmation_slashing_threshold * total_active_balance // 100)
```

Reply:
`confirmation_slashing_threshold` is not intended to be fraction, but rather the maximum amount of stake that the adversary is willing to get slashed.
The idea is that user says "I assume that malicious nodes are willing to spend up to `confirmation_slashing_threshold` ETH to re-org a confirmed block"

This should definitely be clarified somewhere

Reply:
Got it, definitely worth clarification otherwise it is confusing. Absolute value isn't as adaptive to the amount of total stake as a fraction would be, so i'd consider it to be a fraction facilitating better UX by setting this parameter once instead of computing it every time before making a call to the API